src/main/java/com/fasterxml/jackson/core/io/IOContext.java

```diff
@@ -130,25 +133,43 @@ public class IOContext
      *
      * @param src constraints for streaming reads
      * @param swc constraints for streaming writes
-     * @param br BufferRecycler to use, if any ({@code null} if none)
+     * @param erc Error report configuration to use
      * @param contentRef Input source reference for location reporting
      * @param managedResource Whether input source is managed (owned) by Jackson library
+     *
+     * @since 2.16
+     */
+    public IOContext(StreamReadConstraints src, StreamWriteConstraints swc, ErrorReportConfiguration erc,
+                     ContentReference contentRef, boolean managedResource)
+    {
+        this(src, swc, erc, BufferRecyclerPool.acquireBufferRecycler(), contentRef, managedResource);
+    }
+
+    /**
+     * @param src constraints for streaming reads
+     * @param swc constraints for streaming writes
      * @param erc Error report configuration to use
+     * @param br BufferRecycler to use, if any ({@code null} if none)
+     * @param contentRef Input source reference for location reporting
+     * @param managedResource Whether input source is managed (owned) by Jackson library
      *
      * @since 2.16
      */
+    @Deprecated
     public IOContext(StreamReadConstraints src, StreamWriteConstraints swc, ErrorReportConfiguration erc,
-            BufferRecycler br, ContentReference contentRef, boolean managedResource)
+                     BufferRecycler br, ContentReference contentRef, boolean managedResource)
     {
-        _streamReadConstraints = src;
-        _streamWriteConstraints = swc;
+        _streamReadConstraints = (src == null) ?
```

Comment:
Minor nit, probably related to changes in 2.16 in the meantime -- no null checks should be made, caller should pass default values, never null (probably just need to merge/rebase from 2.16)